checks.d/hdfs_datanode.py

```diff
@@ -0,0 +1,169 @@
+'''
+HDFS DataNode Metrics
+---------------------
+hdfs_datanode.dfs_remaining                  The remaining disk space left in bytes
+hdfs_datanode.dfs_capacity                   Disk capacity in bytes
+hdfs_datanode.dfs_used                       Disk usage in bytes
+hdfs_datanode.cache_capacity                 Cache capacity in bytes
+hdfs_datanode.cache_used                     Cache used in bytes
+hdfs_datanode.num_failed_volumes             Number of failed volumes
+hdfs_datanode.last_volume_failure_date       Date the last volume failed
+hdfs_datanode.estimated_capacity_lost_total  The estimated capacity lost in bytes
+hdfs_datanode.num_blocks_cached              The number of blocks cached
+hdfs_datanode.num_blocks_failed_to_cache     The number of blocks that failed to cache
+hdfs_datanode.num_blocks_failed_to_uncache   The number of failed blocks to remove from cache
+'''
+
+# stdlib
+from urlparse import urljoin
+
+# 3rd party
+import requests
+from requests.exceptions import Timeout, HTTPError, InvalidURL, ConnectionError
+from simplejson import JSONDecodeError
+
+# Project
+from checks import AgentCheck
+
+# Service check names
+JMX_SERVICE_CHECK = 'hdfs_datanode.jmx.can_connect'
+
+# URL Paths
+JMX_PATH = 'jmx'
+
+# Metric types
+GAUGE = 'gauge'
+
+# HDFS bean name
+HDFS_DATANODE_BEAN_NAME = 'Hadoop:service=DataNode,name=FSDatasetState'
+
+# HDFS metrics
+HDFS_METRICS = {
+    'Remaining' : ('hdfs_datanode.dfs_remaining',  GAUGE),
+    'Capacity' :('hdfs_datanode.dfs_capacity', GAUGE),
+    'DfsUsed' : ('hdfs_datanode.dfs_used', GAUGE),
+    'CacheCapacity' : ('hdfs_datanode.cache_capacity', GAUGE),
+    'CacheUsed' : ('hdfs_datanode.cache_used', GAUGE),
+    'NumFailedVolumes' : ('hdfs_datanode.num_failed_volumes', GAUGE),
+    'LastVolumeFailureDate' : ('hdfs_datanode.last_volume_failure_date', GAUGE),
+    'EstimatedCapacityLostTotal' : ('hdfs_datanode.estimated_capacity_lost_total', GAUGE),
+    'NumBlocksCached' : ('hdfs_datanode.num_blocks_cached', GAUGE),
+    'NumBlocksFailedToCache' : ('hdfs_datanode.num_blocks_failed_to_cache', GAUGE),
+    'NumBlocksFailedToUnCache' : ('hdfs_datanode.num_blocks_failed_to_uncache', GAUGE)
+}
+
+class HDFSDataNode(AgentCheck):
+
+    def check(self, instance):
+        jmx_address = instance.get('hdfs_datanode_jmx_uri')
+        if jmx_address is None:
+            raise Exception('The JMX URL must be specified in the instance configuration')
+
+        # Get metrics from JMX
+        self._hdfs_datanode_metrics(jmx_address)
+
+    def _hdfs_datanode_metrics(self, jmx_uri):
+        '''
+        Get HDFS data node metrics from JMX
+        '''
+        response = self._rest_request_to_json(jmx_uri,
+            JMX_PATH,
+            query_params={'qry':'Hadoop:service=DataNode,name=FSDatasetState'})
```

Comment:
Could you use the value of `HDFS_DATANODE_BEAN_NAME` here? (unless you have reasons not to)